Mathlib/Algebra/Order/Ring/Archimedean.lean

```diff
@@ -0,0 +1,176 @@
+/-
+Copyright (c) 2025 Weiyi Wang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Weiyi Wang, Violeta Hernández Palacios
+-/
+import Mathlib.Algebra.Order.Archimedean.Class
+import Mathlib.Algebra.Order.Ring.Basic
+import Mathlib.RingTheory.Valuation.Basic
+
+/-!
+# Archimedean classes of a linearly ordered ring
+
+The archimedean classes of a strictly linearly ordered ring can be given the structure of an
+`AddCommMonoid`, by defining
+
+* `0 = mk 1`
+* `mk x + mk y = mk (x * y)`
+
+For a strictly linearly ordered field, we can define a negative as
+
+* `-mk x = mk x⁻¹`
+
+which turns them into a `LinearOrderedAddCommGroupWithTop`.
+
+## Implementation notes
+
+We give Archimedean class an additive structure, rather than a multiplicative one, for the following
+reasons:
+
+* In the ring version of Hahn embedding theorem, `ArchimedeanClassₒ M` naturally becomes
```

Comment:
`ArchimedeanClassₒ` doesn't exist in mathlib (yet). You can mention how `ArchimedeanClass M` can be used as the exponent in `HahnSeries`. `ArchimedeanClassₒ` is meant be a subtype that excludes the top element, so if you want to be precise you can replace this with `{A : ArchimedeanClass M | A \ne \top}`.